Orbit.h

```diff
@@ -9,45 +9,203 @@
+// col: [orbit::Color] User-defined color for the current time slice (see orbit::Color below).
+//
 #define ORBIT_START(name) ORBIT_START_WITH_COLOR(name, orbit::Color::kAuto)
 #define ORBIT_START_WITH_COLOR(name, col) orbit_api::Start(name, col)
 #define ORBIT_STOP() orbit_api::Stop()
 
-// ORBIT_START_ASYNC/ORBIT_STOP_ASYNC: profile time span across threads.
+// ORBIT_START_ASYNC/ORBIT_STOP_ASYNC: Profile time spans across scopes or threads.
+//
+// Overview:
+// Async time spans can be started in one scope and stopped in another. They will be displayed
+// in Orbit on a track uniquely identified by the "name" parameter. Note that those time slices
+// do not represent hierarchical information.
+//
+// Note:
+// We limit the maximum number of characters of the "name" parameter to "kMaxEventStringSize". This
+// limitation may be lifted as we roll out a new dynamic instrumentation implementation.
+// It is possible however to add per-time-slice strings using the ASYNC_STRING macro.
+//
+// Example usage: Tracking "File IO" operations.
+// Thread 1: ORBIT_START_ASYNC("File IO", unique_64_bit_id);  // File IO request site.
+// Thread 1 or 2: ORBIT_ASYNC_STRING(unique_64_bit_id, "My very long file path");
+// Thread 1 or 2: ORBIT_STOP_ASYNC(unique_64_bit_id);  // File IO result site.
+// Result: Multiple time slices labeled with the results of "io_request->GetFileName()" will appear
+//         on a single "async" track named "File IO".
+//
+// Parameters:
+// name: [const char*] Name of the *track* that will display the async events in Orbit.
+// id: [uint64_t] A user-provided unique id for the time slice. This unique id is used to match the
+//     ORBIT_START_ASYNC and ORBIT_STOP_ASYNC calls. An id needs to be unique for the current track.
+// col: [orbit::Color] User-defined color for the current time slice (see orbit::Color below).
+//
 #define ORBIT_START_ASYNC(name, id) ORBIT_START_ASYNC_WITH_COLOR(name, id, orbit::Color::kAuto)
 #define ORBIT_START_ASYNC_WITH_COLOR(name, id, col) orbit_api::StartAsync(name, id, col)
 #define ORBIT_STOP_ASYNC(id) orbit_api::StopAsync(id)
+
+// ORBIT_ASYNC_STRING: Provide an additional string for an async time span.
+//
+// Overview:
+// Provide additional string to be displayed on the time slice corresponding to "id".
+//
+// Note: There is a performance overhead incurred by using the ASYNC_STRING macro. The arbitrarily
+//       long input string will be chunked into substrings of "kMaxEventStringSize" length that will
+//       individually be emitted as multiple profiling events.
+//
+// Example usage: Tracking "File IO" operations.
+// Thread 1: ORBIT_START_ASYNC("File IO", unique_64_bit_id);  // File IO request site.
+// Thread 1 or 2: ORBIT_ASYNC_STRING(unique_64_bit_id, "My very long file path");
+// Thread 1 or 2: ORBIT_STOP_ASYNC(unique_64_bit_id);  // File IO result site.
+// Result: Multiple time slices labeled with the results of "io_request->GetFileName()" will appear
```

Comment:
I don't understand the relation of "io_request->GetFileName()" to the above example. Am I missing something?